[inductor] Add scaled_dot_product_attention to fallback kernels

bertmaher opened the pull request:
Summary:
We don't have decomps/lowerings for SDPA (and probably won't for a
while) so don't warn.

Test Plan: code inspection

Differential Revision: D42878203



cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

drisspg commented on torch/_inductor/lowering.py:
```diff
@@ -1194,6 +1194,8 @@ def apply_constraint(arg, fx_arg):
 make_fallback(aten._fused_moving_avg_obs_fq_helper_functional)
 make_fallback(aten.grid_sampler_2d_backward, require_dense)
 make_fallback(aten.randperm)
+make_fallback(aten._scaled_dot_product_efficient_attention.default)
```
could you also add the flash_attention variants as well?
After this PR: https://github.com/pytorch/pytorch/pull/92917 they will start to show up

eellison commented:
Thanks !

bertmaher commented:
@pytorchbot merge

bertmaher commented:
@pytorchbot merge -f "tests look good enough"

bertmaher committed:
[inductor] Add scaled_dot_product_attention to fallback kernels (#93339)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/93339

We don't have decomps/lowerings for SDPA (and probably won't for a
while) so don't warn.

Test Plan: code inspection

Reviewed By: drisspg

Differential Revision: D42878203

fbshipit-source-id: 1128ad1b6d3539659e65d7d593e5ba9a7d7a3caa
